src/pages/Main/Main.js

```diff
@@ -1,15 +1,478 @@
+      if (filterIds.length) {
+        query += `&${filterType}=${filterIds.join(`&${filterType}=`)}`;
+      }
+    }
+    return query.replace('&', '?');
+  };
+
+  const fetchByFilterQuery = () => {
+    const filterQuery = makeFilterQuery(productfilters);
+    fetch(`${API_URLS.LIST_PAGE}${filterQuery}`)
+      .then(result => result.json())
+      .then(ProductListData => {
+        const sortedProductList = sortProductList(
+          SortTypeState,
+          ProductListData.result
+        );
+        setproductList(sortedProductList);
+      });
+  };
+
+  const sortProductList = (type, productList) => {
+    if (type === 0) {
+      const sortedProductList = productList;
+      sortedProductList.map(products => {
+        products[0].product_options.sort(function (a, b) {
+          return (
+            (a.sell_price === null) - (b.sell_price === null) ||
+            a.sell_price - b.sell_price
+          );
+        });
+      });
+      sortedProductList.sort(function (a, b) {
+        return a[0].product_id - b[0].product_id;
+      });
+      return sortedProductList;
+    } else if (type === 1) {
+      const sortedProductList = productList;
+      sortedProductList.map(products => {
+        products[0].product_options.sort(function (a, b) {
+          return (
+            (a.sell_price === null) - (b.sell_price === null) ||
+            a.sell_price - b.sell_price
+          );
+        });
+      });
+      sortedProductList.sort(function (a, b) {
+        const aPrice = a[0].product_options[0].sell_price;
+        const bPrice = b[0].product_options[0].sell_price;
+        return (aPrice === null) - (bPrice === null) || aPrice - bPrice;
+      });
+      return sortedProductList;
+    } else if (type === 2) {
+      const sortedProductList = productList;
+      sortedProductList.map(products => {
+        products[0].product_options.sort(function (a, b) {
+          return (
+            (a.buy_price === null) - (b.buy_price === null) ||
+            b.buy_price - a.buy_price
+          );
+        });
+      });
+      sortedProductList.sort(function (a, b) {
+        const aPrice = a[0].product_options[0].buy_price;
+        const bPrice = b[0].product_options[0].buy_price;
+        return (aPrice === null) - (bPrice === null) || bPrice - aPrice;
+      });
+      return sortedProductList;
+    }
+  };
+
+  const filterControl = (e, filterTypeQuery) => {
+    if (productfilters[filterTypeQuery].includes(e.target.name)) {
+      setProductFilters(prev => {
+        return {
+          ...prev,
+          [filterTypeQuery]: productfilters[filterTypeQuery].filter(
+            el => el !== e.target.name
+          ),
+        };
+      });
+    } else {
+      setProductFilters(prev => {
+        return {
+          ...prev,
+          [filterTypeQuery]: [
+            ...productfilters[filterTypeQuery],
+            e.target.name,
+          ],
+        };
+      });
+    }
```

Comment:
오재님 중복되는 부분 중에서 변하는 부분만 조건을 걸어서 바꿔주시면 더 간결하고 읽기 쉬운 코드를 작성하실 수 있습니다
```suggestion
    const isAlreadyInFilter = productfilters[filterTypeQuery].includes(e.target.name);
    const newQuery = isAlreadyInFilter ? productfilters[filterTypeQuery].filter(el => el !== e.target.name) : [...productfilters[filterTypeQuery], e.target.name];
    setProductFilters(prev => ({...prev, [filterTypeQuery] : newQuery}))
```

Reply:
넵 참고하겠습니다